### DISCUSSION TITLE
Feature Proposal: Milestones

### DISCUSSION THREAD

**Discussion by cameronmcefee** [TelemetryDeck/Viewer#337] [Ideas]
I recently imported ~5 years of Google Analytics data into another service I'd been using for about a 18 months. I was shocked to see some insane spikes, dropoffs, and other notable features in the newly re-interpreted data that sent me down a rabbit hole of searching through commits, emails, and tweets to try to reconstruct what had happened all those years ago. Now a month or two later, I still see those spikes, but I can't remember what they were for >.<

It might be nice to be able to define important timestamps in a project that could be overlayed onto charts in the time-oriented graphs. Imagine something like this:

![Screenshot 2024-01-18 at 8 18 00 AM](https://github.com/TelemetryDeck/Issues/assets/72919/62e8c56d-7265-4e91-9f46-b34ba1feeea8)

I imagine they could be designed in such a way that they're subtle enough to be ignored, but give you a hover target for more context about what happened on that day.

I could see these being manually edited, or even connected to a repo to display releases/tags

**Comment by winsmith** [comment]
I've been thinking of something similar for a while, just never got around to implementing it. Thanks a lot for requesting this. I've created ticket #338 to track the development
